game/groundforces/template.py

```diff
@@ -0,0 +1,442 @@
+        unit_types = [
+            t
+            for t in self.unit_types
+            if not faction_sensitive or faction.has_access_to_unit_type(t)
+        ]
+
+        alternative_types = []
+        for accessible_unit in faction.accessible_units:
+            if accessible_unit.unit_class in self.unit_classes:
+                unit_types.append(accessible_unit.dcs_id)
+            if accessible_unit.unit_class in self.alternative_classes:
+                alternative_types.append(accessible_unit.dcs_id)
+
+        if not unit_types and not alternative_types and not self.optional:
+            raise StopIteration
+
+        types = unit_types or alternative_types
+        self.set_possible_types(types)
+        return len(types) > 0
+
+    @property
+    def should_be_generated(self) -> bool:
+        if self.optional:
+            return self._enabled
+        return True
+
+    def enable(self) -> None:
+        self._enabled = True
+
+    def disable(self) -> None:
+        self._enabled = False
+
+    def set_enabled(self, enabled: bool) -> None:
+        self._enabled = enabled
+
+    def is_enabled(self) -> bool:
+        return self._enabled
+
+    def set_unit_type(self, unit_type: str) -> None:
+        self._unit_type = unit_type
+
+    def set_possible_types(self, unit_types: list[str]) -> None:
+        self._possible_types = unit_types
+
+    def can_use_unit(self, unit_type: UnitType[Any]) -> bool:
+        return (
+            self.can_use_unit_type(unit_type.dcs_id)
+            or unit_type.unit_class in self.unit_classes
+        )
+
+    def can_use_unit_type(self, unit_type: str) -> bool:
+        return unit_type in self.unit_types
+
+    def reset_unit_counter(self) -> None:
+        count = len(self.units)
+        if self.unit_count:
+            if len(self.unit_count) == 1:
+                count = self.unit_count[0]
+            else:
+                count = random.choice(range(min(self.unit_count), max(self.unit_count)))
+        self._unit_counter = count
+
+    @property
+    def unit_type(self) -> str:
+        if self._unit_type:
+            # Forced type
+            return self._unit_type
+        unit_types = self._possible_types or self.unit_types
+        if unit_types:
+            # Random type
+            return random.choice(unit_types)
+        raise RuntimeError("TemplateGroup has no unit_type")
+
+    @property
+    def size(self) -> int:
+        if not self._unit_counter:
+            self.reset_unit_counter()
+        assert self._unit_counter is not None
+        return self._unit_counter
+
+    @property
+    def max_size(self) -> int:
+        return len(self.units)
+
+    def use_unit(self) -> None:
+        if self._unit_counter is None:
+            self.reset_unit_counter()
+        if self._unit_counter and self._unit_counter > 0:
+            self._unit_counter -= 1
+        else:
+            raise IndexError
```

Comment:
I think everything related to `_unit_counter` can just be a local in `generate_units`